packages/curve-ui-kit/src/features/connect-wallet/lib/utils/wallet-helpers.ts

```diff
@@ -23,3 +25,12 @@ export function getRpcProvider(wallet: Wallet): EIP1193Provider {
   }
   return wallet.provider
 }
+
+const timeout = (message: string, timeoutMs: number) =>
+  new Promise<never>((_, reject) => setTimeout(() => reject(new Error(message)), timeoutMs))
+
+export const withTimeout = <T>(
```

Comment:
Not important but could later be moved to a more generic location. I can see this being useful outside the context of wallet connections.

Reply:
yeah I didn't find a better file so I figured we can move when we have the use for it